src/instructlab/schema/v3/knowledge.json

```diff
@@ -76,28 +76,32 @@
             "description": "The knowledge documents.",
             "type": "object",
             "required": [
```

Comment:
I don't think you have properly specified the [oneOf choice here correctly in JSON schema](https://json-schema.org/understanding-json-schema/reference/combining#oneOf). It is one of either repo+commit or folder.